File: src/main/java/com/dh/ondot/schedule/infra/outbox/OutboxRetryScheduler.java
package com.dh.ondot.schedule.infra.outbox;

import lombok.RequiredArgsConstructor;
import org.springframework.scheduling.annotation.Scheduled;
import org.springframework.stereotype.Component;

@Component
@RequiredArgsConstructor
public class OutboxRetryScheduler {
    private final OutboxBatchDispatcher dispatcher;

    @Scheduled(fixedDelay = 60_000)
    public void processPendingOutboxMessages() {
        dispatcher.dispatchRetryBatch();
        dispatcher.dispatchPendingBatch();
    }
}

File: src/main/kotlin/com/dh/ondot/schedule/infra/outbox/OutboxRetryScheduler.kt
package com.dh.ondot.schedule.infra.outbox

import org.springframework.scheduling.annotation.Scheduled
import org.springframework.stereotype.Component

@Component
class OutboxRetryScheduler(
    private val dispatcher: OutboxBatchDispatcher,
) {
//    @Scheduled(fixedDelay = 60_000)
    fun processPendingOutboxMessages() {
        dispatcher.dispatchRetryBatch()
        dispatcher.dispatchPendingBatch()
    }
}
